packages/firestore/test/integration/api/query.test.ts

```diff
@@ -2106,16 +2107,18 @@ apiDescribe('Queries', persistence => {
               snapshot => snapshot.ref
             );
 
-            // Delete 50 of the 100 documents. Use a WriteBatch, rather than
-            // deleteDoc(), to avoid affecting the local cache.
+            // Delete 50 of the 100 documents. Use a different Firestore
+            // instance to avoid affecting the local cache.
             const deletedDocumentIds = new Set<string>();
-            const writeBatchForDocumentDeletes = writeBatch(db);
-            for (let i = 0; i < createdDocuments.length; i += 2) {
-              const documentToDelete = createdDocuments[i];
-              writeBatchForDocumentDeletes.delete(documentToDelete);
-              deletedDocumentIds.add(documentToDelete.id);
-            }
-            await writeBatchForDocumentDeletes.commit();
+            await withTestDb(PERSISTENCE_MODE_UNSPECIFIED, async db2 => {
```

Comment:
wouldn't this be returning the same Firestore instance(db === db2)? How it would help us avoid affecting cache?

Reply:
Good question (I had to look into this too). But the answer is no because `withTestDb()` creates and returns a newly-created, distinct `Firestore` instance. Therefore, `db` and `db2` have distinct caches and do not share anything, and using `db2` to delete/update documents does not affect the local cache of `db`.

Here is the definition of `withTestDb()`:

https://github.com/firebase/firebase-js-sdk/blob/223ac18f705626aa21791ecfd6655fc2a6df83fe/packages/firestore/test/lite/helpers.ts#L57-L61

which calls `withTestDbSettings()`:

https://github.com/firebase/firebase-js-sdk/blob/223ac18f705626aa21791ecfd6655fc2a6df83fe/packages/firestore/test/lite/helpers.ts#L43-L55

which creates a brand new `FirestoreApp` with a unique name (the `appCount` counter increments for each invocation) and then creates a brand new `Firestore` object from it.

Reply:
Interesting. So they are instances to a same firestore, but from different apps. 
```const app = initializeApp(
    { apiKey: 'fake-api-key', projectId },
    'test-app-' + appCount++
  );```

Reply:
Yep, exactly.